src/update_handler.c

```diff
@@ -1246,6 +1249,9 @@ static gboolean img_to_boot_emmc_handler(RaucImage *image, RaucSlot *dest_slot,
 	if (!res && part_slot)
 		r_emmc_force_part_ro(part_slot->device, NULL);
 
+	if ((out_fd > 0) && (close(out_fd) == -1))
+		g_error("Close output device failed: %s", strerror(errno));
```

Comment:
g_error() is configured to abort() and core-dump. Thus it should not be used for handling possible runtime issues.

I would propose doing an uncontidional `close(out_fd)` in the error handling path of `fsync()` above and a normal `close(out_fd)` with normal error handling here.
This will prevent you from overriding a previous error caused by `fsync()` with a follow-up error caused by a failed `close()` while still having proper error reporting for both cases.